Suport find_albums()

okjuan opened the pull request:
Resolves #13

okjuan committed:
Update var/func names to allow songs AND albums
- Also, trivial: tweak reason string in test skips

okjuan committed:
Update broken integration test

okjuan committed:
Remove req for bearer token from integration tests

okjuan committed:
Loosen assert in integration test
This one seems to have changed on me... maybe spotify search is not deterministic

okjuan committed:
Support find_album like find_song
- Also, update song metadata stripping:
    - Broaden def'n of name metadata stripping to support albums
    - Treat parentheses as containing metadata e.g. 'White Album (Remastered)'
    - Loosen name-matching requirements so that we only depend on metadata stripping when case-insensitive exact match fails
    - Remove metadata from songs and album with different logic

okjuan committed:
Add shell script

okjuan committed:
Album input tests

okjuan committed:
Fail spotify client integration tests that need manual check

IsaacSahle commented on song_scrounger/models/album.py:
```diff
@@ -0,0 +1,18 @@
+        """
+        self.name = name
+        self.spotify_uri = spotify_uri
+        self.artists = artists
```
super-nit: you could avoid assigning None to self.popularity by changing the logic a bit... something like

if p is not None and (p is invalid):
  raise ...
self.p = p

IsaacSahle commented on song_scrounger/models/album.py:
```diff
@@ -0,0 +1,18 @@
+class Album():
+    def __init__(self, name, spotify_uri, artists, popularity=None):
```
could you assign a default value of int maybe (-1) to avoid None check below?

okjuan replied:
I'll leave this is in for now because I'm making `popularity` a hard req

okjuan committed:
Publish pkg v0.2.0 w/ find_albums functionality
- However, albums don't contain track info
- Moreover, can't create playlists from album spotify URI yet

okjuan committed:
Convert spotify lib models to songscrounger models with spotify client

okjuan committed:
Get album tracklist
- Skip  checks now that additional  call is made for album songs
